chain/chain/src/types.rs

```diff
@@ -306,6 +304,21 @@ pub trait RuntimeAdapter: Send + Sync {
         }
         Ok(receipts_hashes)
     }
+
+    // TODO return Result?
+    fn collect_receipts(&self, receipt_proofs: &Vec<ReceiptProof>) -> Vec<Receipt> {
+        receipt_proofs.iter().map(|x| x.0.clone()).flatten().collect()
+    }
+
+    // TODO return Result?
+    fn collect_receipts_from_response(
```

Comment:
same - why does it need to be on runtime adapter? seems like standalone function would work as well?

Reply:
++